docs/source/fx.rst

```diff
@@ -10,7 +10,261 @@ Overview
+
+    traced_without_activation = torch.fx.symbolic_trace(without_activation)
+    print(traced_without_activation.code)
+    """
+    def forward(self, x):
+        linear_1 = self.linear(x);  x = None
+        return linear_1
+    """
+
+    traced_with_activation = torch.fx.symbolic_trace(with_activation)
+    print(traced_with_activation.code)
+    """
+    import torch
+    def forward(self, x):
+        linear_1 = self.linear(x);  x = None
+        relu_1 = torch.relu(linear_1);  linear_1 = None
+        return relu_1
+    """
+
+The if-statement ``if self.do_activation`` does not depend on any
+function inputs, thus it is static. ``do_activation`` can be considered
+to be a hyper-parameter, and the traces of different instances of
+``MyModule`` with different values for that parameter have different
+code. This is a valid pattern that is supported by symbolic tracing.
+
+Many instances of dynamic control flow are semantically static control
+flow. These instances can be made to support symbolic tracing by
+removing the data dependencies on input values, for example by moving
+values to ``Module`` attributes.
+
+Non-\ ``torch`` Functions
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+FX uses ``__torch_function__`` as the mechanism by which it intercepts
+calls (see the `technical
+overview <https://github.com/pytorch/pytorch/blob/master/torch/fx/OVERVIEW.md#technical-details>`__
+for more information about this). Some functions, such as builtin Python
+functions or those in the ``math`` module, are things that are not
+covered by ``__torch_function__``, but we would still like to capture
+them in symbolic tracing. For example:
+
+::
+
+    from math import sqrt
+
+    def normalize(x):
+        """
+        Normalize `x` by the size of the batch dimension
+        """
+        return x / sqrt(len(x))
+
+    # It's valid Python code
+    normalize(torch.rand(3, 4))
+
+    traced = torch.fx.symbolic_trace(normalize)
+    """
+      <...>
+      File "sqrt.py", line 9, in normalize
+        return x / sqrt(len(x))
+      File "pytorch/torch/fx/proxy.py", line 161, in __len__
+        raise RuntimeError("'len' is not supported in symbolic tracing by default. If you want "
+    RuntimeError: 'len' is not supported in symbolic tracing by default. If you want this call to be recorded, please call torch.fx.wrap('len') at module scope
+    """
+
+The error tells us that the built-in function ``len`` is not supported.
+We can make it so that functions like this are recorded in the trace as
+direct calls using the :func:`wrap` API:
+
+::
+
+    torch.fx.wrap('len')
+    torch.fx.wrap('sqrt')
+
+    traced = torch.fx.symbolic_trace(normalize)
+
+    print(traced.code)
+    """
+    import math
+    def forward(self, x):
+        len_1 = len(x)
+        sqrt_1 = math.sqrt(len_1);  len_1 = None
+        truediv = x / sqrt_1;  x = sqrt_1 = None
+        return truediv
+    """
+
+Customizing Tracing with the ``Tracer`` class
```

Comment:
I think this is a confusing segue. `Customizing Tracer with the tracer class` should be a solution to a limitation, it shouldn't be at the top level.

Reply:
I wanted it to be top-level so that Tracer documentation is in one place and any other section of the docs can reference the single source of truth. I can see if i can make these references more clear